tests/test_forms.py

```diff
@@ -33,3 +37,330 @@ class Meta:
         article.refresh_from_db()
         self.assertEqual(article.title_nl, "Nederlandse taal")
         self.assertEqual(article.title_en, "English language")
+
+
+class Form(TranslationModelForm):
+    """Challenge form."""
+
+    start_date = forms.DateField(label="start date")
+    end_date = forms.DateField(label="end date")
+
+    class Meta:
+        model = Challenge
+        fields = ["title", "start_date", "default_language", "header"]
+        required = {"title": True, "header": False}
+
+
+class ExcludeForm(TranslationModelForm):
+    """Challenge form."""
+
+    start_date = forms.DateField(label="start date")
+    end_date = forms.DateField(label="end date")
+
+    field_order = ["start_date", "title", "end_date"]  # test the custom field_order setting
+
+    class Meta:
+        model = Challenge
+        exclude = ["header", "default_language"]
+        languages = ["browser", "de", "fallback"]
+        fallback_language = "fr"
+        widgets = {"title": forms.widgets.Textarea}
+
+
+class TranslationFormTestCase(TestCase):
+    def test_languages_errors(self):
+        """Test the error messages for incorrect languages options."""
+
+        with self.assertRaisesMessage(ValueError, "languages: values should be strings"):
+            Form(languages=[0, "fallback"])
+
+        with self.assertRaisesMessage(ValueError, "languages: value es_it is not permitted"):
+            Form(languages=["es_it", "fallback"])
+
+        with self.assertRaisesMessage(ValueError, "languages: value e is not permitted"):
+            Form(languages=["e", "fallback"])
+
+        with self.assertRaisesMessage(
+            ValueError, "languages: xx is not an available language in the system"
+        ):
+            Form(languages=["xx", "fallback"])
+
+        class NoLanguageForm(TranslationModelForm):
+            """Challenge form where fallback can be edited."""
+
+            class Meta:
+                model = Challenge
+                fields = ["title", "header"]
+                languages = []
+
+        with self.assertRaisesMessage(
+            ValueError, "languages: Error. No languages have been defined."
+        ):
+            NoLanguageForm()
+
+    def test_defaults(self):
+        """Test the default form options."""
+        form = Form()
+        self.assertEqual(form.languages, ["browser"])
+        self.assertEqual(form.fallback_language, get_default_language())
+
+    def test_get_fallback_language(self):
+        """Test that the form fallback language is set correctly."""
+
+        with self.subTest("Test the default fallback_language"):
+            form = Form()
+            self.assertEqual(form.fallback_language, get_default_language())
```

Comment:
Better to be explicit here:
```suggestion
            self.assertEqual(form.fallback_language, "en")
```